pydicom/sr/codedict.py

```diff
@@ -15,17 +15,12 @@
 
 
 def _filtered(allnames, filters):
-    """Helper function for dir() methods"""
-    matches = {}
-    for filter_ in filters:
-        filter_ = filter_.lower()
-        match = [x for x in allnames if x.lower().find(filter_) != -1]
-        matches.update(dict([(x, 1) for x in match]))
-    if filters:
-        names = sorted(matches.keys())
-        return names
-    else:
-        return sorted(allnames)
+    if not filters:
+        return sorted(set(allnames))
+    lowered_filters = tuple(map(str.lower, filters))
+    include_name = lambda name: any(lowered_filter in name.lower() # noqa: E731
+                                    for lowered_filter in lowered_filters)
+    return sorted(set(filter(include_name, allnames)))
```

Comment:
*pydicom* uses list comprehensions rather than `map` and `filter`, which is, I believe, the  recommended way in general for modern Python.

Reply:
Fixed. Did generator expressions though, no need to keep stuff in memory.